reports.py

```diff
@@ -183,6 +196,54 @@ def parser_assembly_stats(parser=argparse.ArgumentParser()):
 __commands__.append(('assembly_stats', parser_assembly_stats))
 
 
+def _get_samples_from_bam(bam):
+    with pysam.AlignmentFile(bam) as af:
+        return set(rg['SM'] for rg in af.header['RG'])
+def _get_chrs_from_bam(bam):
+    with pysam.AlignmentFile(bam) as af:
+        return list(sq['SN'] for sq in af.header['SQ'])
+
+def parser_coverage_only(parser=argparse.ArgumentParser()):
+    parser.add_argument('mapped_bams', nargs='+', help='Aligned-to-self mapped bam files.')
+    parser.add_argument('out_report', help='Output report file.')
+    parser.add_argument('--cov_thresholds',
+                        nargs='+',
+                        type=int,
+                        default=(1, 5, 20, 100),
+                        help='Genome coverage thresholds to report on. (default: %(default)s)')
+    util.cmd.common_args(parser, (('loglevel', None), ('version', None)))
+    util.cmd.attach_main(parser, coverage_only, split_args=True)
+    return parser
+def coverage_only(mapped_bams, out_report, cov_thresholds=(1, 5, 20, 100)):
+    header = ['sample','aln2self_cov_median', 'aln2self_cov_mean', 'aln2self_cov_mean_non0']
+    header += ['aln2self_cov_%dX' % thresh for thresh in cov_thresholds]
+    with open(out_report, 'wt') as outf:
+        outf.write('\t'.join(header)+'\n')
+        for bam in mapped_bams:
+            # check for index and auto-create if needed
+            with pysam.AlignmentFile(bam) as af:
+                is_indexed = af.has_index()
+            if not is_indexed:
+                pysam.index(bam)
+            # get unique sample name
+            samples = _get_samples_from_bam(bam)
+            if len(samples) != 1:
+                raise Exception("input bam file {} has {} unique samples: {} (require one unique sample)".format(bam, len(samples), str(samples)))
+            sample_name = samples.pop()
+            # get and write coverage stats
+            row = genome_coverage_stats_only(bam, cov_thresholds=cov_thresholds)
```

Comment:
How slow is indexing and pulling the coverage stats? Would it make sense to break out these functions to a process pool?

Reply:
I was surprised how quick it was..  seems to be seconds per sample? https://platform.dnanexus.com/projects/FBv3fq00kyFkqP9J34zxb3gq/monitor/analysis/FBv3xx00kyFY7Kpv28JZpv5J shows under a minute for 15 LASV genomes..